main/forum/forumfunction.inc.php

```diff
@@ -3835,8 +3835,8 @@ function store_reply($current_forum, $values, $courseId = 0, $userId = 0)
             $table_posts,
             [
                 'c_id' => $courseId,
-                'post_title' => $values['post_title'],
-                'post_text' => isset($values['post_text']) ? ($values['post_text']) : null,
+                'post_title' => Security::remove_XSS($values['post_title']),
```

Comment:
There's no need to add Security::remove_XSS before the insert to the DB

Reply:
Why keeping XSS'ed html in database?
If is not displayed, it's not useful, so not needed in db.

Reply:
Security::remove_XSS uses HtmlPurifier and it's very agressive. In some cases it will remove valid HTML content.

It's better to use remove_XSS before displaying because we can edit the "aggressiveness" of the clean in order to allow some content.

Reply:
Seems Legit. Thks.
I cleaned the branch and I made a new pull request #3884

Reply:
@renaudlemaire We do this in general in Chamilo, because something stored in the database is not necessarily shown on a web page (it might get processed in a different way, like exporting in PDF, CSV, XML, etc). HTMLPurifier does clean it but sometimes we just might want the original post (even if only in terms of being able to audit it afterwards).
All output from the database to a web page should be filtered against XSS however... always (if you find a place where it's not, it's a mistake and probably a vulnerability).

Reply:
@ywarnier Understood. Thanks.